philosophy_of_software_design/README.md

```diff
@@ -43,3 +43,14 @@ The need for extensive documentation is often a sign that the design isn't
 quite right. Complexity is incremental, it comes from an accumulation of
 dependencies and obscurities.
 
+## 3. 
+
+## 4. Modules Should be Deep
+
+Modular design attempts to minimize the dependencies between the modules that conform a system, trying to keep them as independent as possible from each other.
+We think of each module in two parts:
+  - Interface: it abstracts the module, describing _what_ it does.
+  - Implementation: the code that carries out what the interface states.
+We could also classify modules into two categories:
+  - Deep modules: they provide powerful functionalities yet have simple interfaces.
```

Comment:
elaborate on the deep vs shallow module idea below, this is arguably the best contribution of the book. mention that a shallow module is a red flag (there should be summaries of design principles and red flags at the bottom of this README just like in the book, but that can be done in a separate PR)

mention the definition of abstraction

include the deep vs shallow diagram here